packages/orm/src/index.ts

```diff
@@ -437,3 +438,34 @@ export function getConstructorFromTaggedId(id: string): EntityConstructor<any> {
 function equal(a: any, b: any): boolean {
   return a === b || (a instanceof Date && b instanceof Date && a.getTime() == b.getTime());
 }
+
+// The Entity type doesn't model the `changes` structure yet, so do this for now.
+interface EntityChanges<T> {
+  // Ideally we could use Record<keyof T, { hasChanged: boolean }> but we only want
+  // keys that are actually columns. Right now OptsOf<T> has collections (not in .changed)
+  // and OrderOf<T> has id/createdAt/updatedAt (also not in .changed), so just using an index
+  // type for now.
+  changes: { [key: string]: { hasChanged: boolean } };
+}
+
+const cannotBeChangedRule = Symbol("cannotBeChangedRule");
+
+export function cannotBeChanged<T extends Entity & EntityChanges<T>, K extends keyof T & string>(
+  fieldName: K,
+  unless?: (entity: T) => Promise<boolean>,
+): ValidationRule<T> {
+  const fn: ValidationRule<T> = async (entity) => {
+    if (!entity.isNewEntity && entity.changes[fieldName].hasChanged && (!unless || !(await unless(entity)))) {
+      return `${capitalCase(fieldName)} cannot be updated`;
+    }
+    return undefined;
+  };
+  if (!unless) {
+    (fn as any)[cannotBeChangedRule] = fieldName;
+  }
```

Comment:
question: Is this indicating that if we specify `unless`, then the rule will still be reactive?

Reply:
Yeah, since the reactivity (or lack of reactivity) rules are built once on the initial boot of the system (not at codegen time or at each `em.flush()`), we can't really know for sure if this is always-readonly or always-not-readonly, so have to default to assuming its reactive.

I guess at least for this part of building the initial reactive rule list...

Maybe we could work this conditionally-read-only-ness into the evaluation of the reactive things, but not sure...